### ISSUE TITLE
Add --left-delimiter and --right-delimiter command-line options

### ISSUE THREAD

**Issue by WolverineFan** [hashicorp/consul-template#1008]
I'd like to be able to pass in the left/right delimiter on the command-line.

### Consul Template version
0.18.0 - 0.19.2

### Configuration
I can override the template delimiters in a config file using 
```hcl
template {
  ...
  left_delimiter  = "{:"
  right_delimiter = ":}"
  ...
}
```

But I have some validation code that renders the template to a temp file:
```bash
consul-template -template "$1:/tmp/consul-template-rendered" -once
```
This doesn't work if I override the delimiter.  If I could pass the delimiters on the command-line it would.  Another option would be to allow overriding the delimiters globally in addition to in the `template` block.  Then I believe I could have consul-template read a config file with just global configuration while still passing `-template ... -once` for my validation.

**Comment by sethvargo** [comment]
Hi @WolverineFan 

Unfortunately it's not possible to do this via the CLI because it's a _template-specific_ option, not a global option. Template-specific options require a configuration file.

**Comment by WolverineFan** [comment]
@sethvargo What about my suggestion of making it a global option?  Honestly either one would solve the problem...

**Comment by WolverineFan** [comment]
Let me rephrase...  Allow setting a global `default` value and/or a CLI option to set the global default.

**Comment by WolverineFan** [comment]
@sethvargo I noticed last night that `wait` is already an option you can configure a global default for.  I'm think delimiters are a reasonable thing to treat the same way.  I tried digging in to the source but I can't quite figure out how the `wait` default values get populated and used by the `template` block.  I'm happy to do the work (or at least some of it) but I'd need some tips to get started.

**Comment by sethvargo** [comment]
Hi @WolverineFan 

`wait` is a legacy option, so we keep it for backwards compatibility. It predates template-specific configurations.

**Comment by WolverineFan** [comment]
@sethvargo So you see no value in allowing a global default for left/right delimiter?  I'm surprised I'm the only person interested...

**Comment by dd-caleb** [comment]
I also found this super-confusing, didn't realize the template option is for a single template and not global options for all possible `-template` on the command line. Because of this there doesn't appear to be an easy way to change the delimiters across the board.
